src/ensembl/src/content/app/entity-viewer/EntityViewer.tsx

```diff
@@ -46,6 +51,9 @@ const EntityViewer = (props: Props) => {
 
   useEffect(() => {
     props.setDataFromUrl(params);
+    if (params.entityId) {
+      props.updateEnsObject(params.entityId);
+    }
```

Comment:
should this perhaps be the part of how the reducer reacts to the `setDataFromUrl ` action?